lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb

```diff
@@ -402,7 +402,14 @@ def validate_after_scope_change?
           else
             all_records = @subject.class.all
             @options[:scopes].all? do |scope|
-              previous_value = all_records.map(&scope).compact.max
+              previous_value =
+                if is_a_boolean_scope?(all_records, scope)
+                  value = all_records.map(&scope).last
+                  @existing_record.destroy
```

Comment:
When we are dealing with date, string and integers for example we do not have this problem because the `next_value` is always a unused value.

```
all_records
=> [#<Example:0x007f9a932a56b8 id: 1, attr: "dummy value", scope: 2>, #<Example:0x007f9a932a53e8 id: 2, attr: "dummy value", scope: 3>]

previous_value
=> 3

next_value
=> 4

allows_value_of(existing_value, @expected_message)
=> true
```

But when we are dealing with a boolean our `next_value` will always be equal the first one created by the matcher

```
all_records
=> [#<Example:0x007f9a96982d48 id: 1, attr: "dummy value", scope: false>, #<Example:0x007f9a969825c8 id: 2, attr: "dummy value", scope: true>]

previous_value
=> true

next_value
=> false

allows_value_of(existing_value, @expected_message)
=> true
```

So if I not delete the first record, the `allows_value_of` will always be `false` since the `scope` is already defined on the first record, and here we can define only two value, `true` and `false`.

I really don't know if this is the better approach, if you have any idea let me know. I will dig around to see if I found another way to do that.